backend/src/jv2backend/requestData.py

```diff
@@ -0,0 +1,136 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+# Copyright (c) 2023 Team JournalViewer and contributors
+
+import typing
+import jv2backend.journals
+from jv2backend.utils import url_join
+
+
+class RequestData:
+    """Simple class for checking / handling POST data supplied to a Flask
+    route"""
+
+    def __init__(self, requestData: typing.Any,
+                 library: jv2backend.journals.JournalLibrary,
+                 require_in_library=False,
+                 require_data_directory=False,
+                 require_filename=False) -> None:
+        """Set up the class. The POST data contains the following:
+              rootUrl: The main, root URL path (http or file)
+            directory: [OPTIONAL] Directory within the rootUrl to consider
+        dataDirectory: [OPTIONAL] Associated run data directory
+             filename: [OPTIONAL] Target filename
+
+        We can make various stipulations on the available data:
+            require_in_library: Whether a library matching the library_key
+                                must already exist
+        require_data_directory: Whether a 'dataDirectory' must be provided
+              require_filename: Whether a 'filename' must be provided
+        """
+        self._root_url: str = None
+        self._directory: str = None
+        self._full_url: str = None
+        self._data_directory: str = None
+        self._filename: str = None
+        self._journal_collection: jv2backend.journals.JournalCollection = None
+        self._is_http: bool = False
+        self._error: str = None
+
+        # Root url is always required
+        if "rootUrl" not in requestData:
+            self._error = "No root URL provided in request."
+            return
+        self._root_url = requestData["rootUrl"]
+
+        # Determine whether this is a network or file url
+        self._is_http = (self._root_url.lower().startswith("http://") or
+                         self._root_url.lower().startswith("https://"))
```

Comment:
Maybe knowing whether or not something is a URL or filepath is better left to the frontend/requester? It would be safer if the request included a parameter of 'sourcetype' which can be set to URL or Path for example. I'd assume that the frontend would be able to get this information easily since it has to get the user to select a file if they wish.

Reply:
It's an either / or question - either the backend needs to know and the frontend has no idea, or vice versa. As you suggest it might make more sense for the type of data source to be known at the frontend level, since that's a defined property already. I'll capture this in an issue for future addressing.